components/opponent/wikis/trackmania/opponent_display_custom.lua

```diff
@@ -0,0 +1,223 @@
+			self:createTeam(opponent.template or 'tbd', options)
+		elseif opponent.type == Opponent.solo or opponent.type == Opponent.duo then
+			self:createPlayers(opponent)
+		elseif opponent.type == Opponent.literal then
+			self:createLiteral(opponent.name or '')
+		end
+
+		self.root = html.create('div'):addClass('brkts-opponent-entry')
+			:node(self.content)
+	end
+)
+
+OpponentDisplayCustom.BracketOpponentEntry.createTeam = OpponentDisplay.BracketOpponentEntry.createTeam
+
+function OpponentDisplayCustom.BracketOpponentEntry:createPlayers(opponent)
+	local players = opponent.players
+	if #players == 1 then
+		local playerNode = PlayerDisplay.BlockPlayer({
+			player = players[1],
+			overflow = 'ellipsis',
+		})
+		self.content:node(playerNode)
+	else
+		local playersNode = OpponentDisplayCustom.PlayerInlineOpponent{
+			opponent = opponent
+		}
+
+		self.content:node(playersNode)
+	end
+end
+
+OpponentDisplayCustom.BracketOpponentEntry.createLiteral = OpponentDisplay.BracketOpponentEntry.createLiteral
+
+function OpponentDisplayCustom.BracketOpponentEntry:addScores(opponent)
+	local extradata = opponent.extradata or {}
+	if not extradata.additionalScores then
+		OpponentDisplay.BracketOpponentEntry.addScores(self, opponent)
+	else
+		local score1Node = OpponentDisplay.BracketScore({
+			isWinner = extradata.set1win,
+			scoreText = OpponentDisplay.InlineScore(opponent),
+		})
+		self.root:node(score1Node)
+
+		local score2Node
+		if opponent.extradata.score2 or opponent.score2 then
+			score2Node = OpponentDisplay.BracketScore({
+				isWinner = extradata.set2win,
+				scoreText = OpponentDisplayCustom.InlineScore2(opponent),
+			})
+		end
+		self.root:node(score2Node)
+
+		local score3Node
+		if opponent.extradata.score3 then
+			score3Node = OpponentDisplay.BracketScore({
+				isWinner = extradata.set3win,
+				scoreText = OpponentDisplayCustom.InlineScore3(opponent),
+			})
+		end
+		self.root:node(score3Node)
+
+		if (opponent.placement2 or opponent.placement or 0) == 1
+			or opponent.advances then
+			self.content:addClass('brkts-opponent-win')
+		end
+	end
+end
+
+--[[
+Displays the second score or status of the opponent, as a string.
+]]
+function OpponentDisplayCustom.InlineScore2(opponent)
+	local score2 = opponent.extradata.score2
+	if opponent.status2 == 'S' then
+		if opponent.score2 == 0 and Opponent.isTbd(opponent) then
+			return ''
+		else
+			return opponent.score2 ~= -1 and tostring(opponent.score2) or ''
+		end
+	else
+		return opponent.status2 or score2 or ''
+	end
+end
+
+--[[
+Displays the third score or status of the opponent, as a string.
+]]
+function OpponentDisplayCustom.InlineScore3(opponent)
+	local score3 = opponent.extradata.score3
+	if opponent.status3 == 'S' then
+		if opponent.score3 == 0 and Opponent.isTbd(opponent) then
+			return ''
+		else
+			return opponent.score3 ~= -1 and tostring(opponent.score3) or ''
+		end
+	else
+		return opponent.status3 or score3 or ''
+	end
+end
```

Comment:
needs testing!!!

but i think this would be a bit cleaner compared to the current version

```suggestion
local SCORE_STATUS = 'S'
local NO_SCORE = -1
local ZERO_SCORE = 0

...

function OpponentDisplayCustom.BracketOpponentEntry:addScores(opponent)
	local extradata = opponent.extradata or {}
	if not extradata.additionalScores then
		OpponentDisplay.BracketOpponentEntry.addScores(self, opponent)
	else
		local score1Node = OpponentDisplay.BracketScore({
			isWinner = extradata.set1win,
			scoreText = OpponentDisplayCustom.InlineScore(opponent, ''),
		})
		self.root:node(score1Node)

		local score2Node
		if opponent.extradata.score2 or opponent.score2 then
			score2Node = OpponentDisplay.BracketScore({
				isWinner = extradata.set2win,
				scoreText = OpponentDisplayCustom.InlineScore(opponent, 2),
			})
		end
		self.root:node(score2Node)

		local score3Node
		if opponent.extradata.score3 then
			score3Node = OpponentDisplay.BracketScore({
				isWinner = extradata.set3win,
				scoreText = OpponentDisplayCustom.InlineScore(opponent, 3)
			})
		end
		self.root:node(score3Node)

		if (opponent.placement2 or opponent.placement or 0) == 1
			or opponent.advances then
			self.content:addClass('brkts-opponent-win')
		end
	end
end

function OpponentDisplayCustom.InlineScore(opponent, scoreIndex)
	scoreIndex = scoreIndex or ''

	local status = opponent['status' .. scoreIndex] or opponent.extradata['status' .. scoreIndex] or ''
	local score = tonumber(opponent['score' .. scoreIndex] or opponent.extradata['score' .. scoreIndex]) or 0

	if status == SCORE_STATUS then
		if (score == ZERO_SCORE and Opponent.isTbd(opponent)) or score == NO_SCORE then
			return ''
		else
			return score
		end
	end

	return status
end
```

Reply:
https://github.com/Liquipedia/Lua-Modules/pull/2262/commits/07d7410c8ba146279845c2d699b6acdc52cb3da3